aiosparql/client.py

```diff
@@ -178,6 +185,57 @@ def _pretty_print_query(self, query: str) -> str:
                     explanation=explanation)
             return await resp.json()
 
+    def _crud_request(self, method, graph=None, data=None, accept=None,
+                      content_type=None):
+        if not self.crud_endpoint:
+            raise ValueError("CRUD endpoint not specified")
+        url = self.crud_endpoint
+        if graph:
+            params = {'graph': graph.value}
+        elif self.graph:
+            params = {'graph': self.graph.value}
+        else:
+            params = None
+            url += "?default"
+        headers = {}
+        if content_type:
+            headers["Content-Type"] = content_type
+        if accept:
+            headers["Accept"] = accept
+        logger.debug("Sending %s request to CRUD endpoint %s with headers "
+                     "%r, and params %r", method, url, headers, params)
+        return self.session.request(method, url, params=params,
+                                    headers=headers, data=data)
+
+    def get(self, *, format: str, graph: Optional[IRI] = None):
+        return self._crud_request("GET", graph=graph, accept=format)
+
+    async def put(self, data: Union[bytes, IOBase], *, format: str,
+                  graph: Optional[IRI] = None):
+        async with self._crud_request("PUT", graph=graph, data=data,
+                                      content_type=format) as resp:
+            resp.raise_for_status()
+
+    async def delete(self, graph: Optional[IRI] = None):
+        async with self._crud_request("DELETE", graph=graph) as resp:
+            resp.raise_for_status()
+
+    async def post(self, data: Union[bytes, IOBase], *, format: str,
+                   graph: Optional[IRI] = None):
+        async with self._crud_request("POST", graph=graph, data=data,
+                                      content_type=format) as resp:
+            resp.raise_for_status()
+
+    def head(self, *, format: str, graph: Optional[IRI] = None):
+        return self._crud_request("HEAD", graph=graph, accept=format)
+
+    async def patch(self, data: Union[bytes, IOBase], *,
+                    graph: Optional[IRI] = None):
+        async with self._crud_request("PATCH", graph=graph, data=data,
+                                      accept="application/json") as resp:
+            resp.raise_for_status()
+            return await resp.json()
```

Comment:
I'm not sure about this code. According to the specs, the PATCH method is like an endpoint for SPARQL 1.1 update:

> SPARQL 1.1 Update can be used as a patch document. 

I suppose that without being able to really test it I should simply remove the code for now or raise NotImplementedError.